packages/snaps-utils/src/snaps.ts

```diff
@@ -232,25 +239,32 @@ export function validateSnapShasum(
 
 export const LOCALHOST_HOSTNAMES = ['localhost', '127.0.0.1', '[::1]'] as const;
 
+// Require snap ids to only consist of printable ASCII characters
+export const BaseSnapIdStruct = pattern(string(), /^[\x21-\x7E]*$/u);
```

Comment:
How about [valid NPM names](https://www.npmjs.com/package/validate-npm-package-name) instead, those rules were forged in fire

Reply:
Unfortunately we cant. Our local snap IDs are not compatible with the NPM naming scheme. They will fail this check: https://github.com/npm/validate-npm-package-name/blob/main/lib/index.js#L74